contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -398,42 +400,83 @@ contract InterchainTokenService is
         );
     }
 
+    // Returns the amount of token that this call is worth. If `tokenAddress` is `0`, then value is in terms of the native token, otherwise it's in terms of the token address.
+    function contractCallValue(
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) public view virtual onlyRemoteService(sourceChain, sourceAddress) notPaused returns (address, uint256) {
+        (, bytes32 tokenId, , uint256 amount) = abi.decode(payload, (uint256, bytes32, bytes, uint256));
+
+        ITokenManager tokenManager = ITokenManager(getValidTokenManagerAddress(tokenId));
+        return (tokenManager.tokenAddress(), amount);
+    }
+
+    function expressExecute(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceService,
+        bytes calldata payload
+    ) external payable notPaused {
+        uint256 selector = abi.decode(payload, (uint256));
+        if (selector != SELECTOR_RECEIVE_TOKEN && selector != SELECTOR_RECEIVE_TOKEN_WITH_DATA) {
+            revert InvalidExpressSelector();
+        }
+
+        bytes32 payloadHash = keccak256(payload);
+        _expressReceiveToken(commandId, sourceChain, payload, payloadHash);
+
+        emit ExpressExecuted(commandId, sourceChain, sourceService, payloadHash, msg.sender);
+    }
+
+    function getExpressExecutor(
+        bytes32 commandId,
+        string calldata /*sourceChain*/,
+        string calldata /*sourceAddress*/,
+        bytes32 payloadHash
+    ) external view returns (address) {
+        return _getExpressCaller(commandId, payloadHash);
+    }
+
     /**
      * @notice Uses the caller's tokens to fullfill a sendCall ahead of time. Use this only if you have detected an outgoing
      * interchainTransfer that matches the parameters passed here.
      * @dev This is not to be used with fee on transfer tokens as it will incur losses for the express caller.
      * @param payload the payload of the receive token
      * @param commandId the sendHash detected at the sourceChain.
      */
-    function expressReceiveToken(bytes calldata payload, bytes32 commandId, string calldata sourceChain) external {
-        if (gateway.isCommandExecuted(commandId)) revert AlreadyExecuted(commandId);
+    function _expressReceiveToken(bytes32 commandId, string calldata sourceChain, bytes calldata payload, bytes32 payloadHash) internal {
```

Comment:
Can we just reuse ExpressExecutorTracker contract for storing express info instead of redefining with express caller? We can move the interface to the individual express executables so it doesn't interfere with the inheritance
https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/express/ExpressExecutorTracker.sol#L7

At the least we should use express executor naming convention

Reply:
Having issues with contract size:
```
Transaction reverted: trying to deploy a contract whose code is too large
```

Reply:
Resolved with setting lower optimizer number